cdm/core/src/main/java/ucar/nc2/ft/point/writer2/WriterCFStationCollection.java

```diff
@@ -96,36 +99,35 @@ void writeHeader(List<StationFeature> stns, StationPointFeature spf) throws IOEx
         desc_strlen = Math.max(desc_strlen, stn.getDescription().length());
       if (stn.getWmoId() != null)
         wmo_strlen = Math.max(wmo_strlen, stn.getWmoId().length());
-    }
-
-    llbb = CFPointWriterUtils.getBoundingBox(stnList); // gets written in super.finish();
 
-    StationFeature sf = spf.getStation();
-    StructureData stnData = sf.getFeatureData();
-    StructureData obsData = spf.getFeatureData();
-
-    List<VariableSimpleIF> coords = new ArrayList<>();
-    coords.add(VariableSimpleBuilder.makeScalar(timeName, "time of measurement", timeUnit.getUdUnit(), DataType.DOUBLE)
-        .addAttribute(CF.CALENDAR, timeUnit.getCalendar().toString()).build());
+      if (stn instanceof DsgFeatureCollection) {
+        DsgFeatureCollection dsgStation = (DsgFeatureCollection) stn;
+        if (coords.stream().noneMatch(x -> x.getShortName().equals(dsgStation.getTimeName()))) {
+          coords.add(VariableSimpleBuilder
+              .makeScalar(dsgStation.getTimeName(), "time of measurement", dsgStation.getTimeUnit().getUdUnit(),
+                  DataType.DOUBLE)
+              .addAttribute(CF.CALENDAR, dsgStation.getTimeUnit().getCalendar().toString()).build());
+        }
+      } else {
+        coords.add(
+            VariableSimpleBuilder.makeScalar(timeName, "time of measurement", timeUnit.getUdUnit(), DataType.DOUBLE)
+                .addAttribute(CF.CALENDAR, timeUnit.getCalendar().toString()).build());
+      }
+    }
 
+    llbb = ucar.nc2.ft.point.writer.CFPointWriterUtils.getBoundingBox(stnList); // gets written in super.finish();
     coords.add(VariableSimpleBuilder
         .makeScalar(stationIndexName, "station index for this observation record", null, DataType.INT)
         .addAttribute(CF.INSTANCE_DIMENSION, stationDimName).build());
 
-    Formatter coordNames = new Formatter().format("%s %s %s", timeName, latName, lonName);
-    if (useAlt)
-      coordNames.format(" %s", stationAltName);
-
-    super.writeHeader(coords, stnData, null, obsData, coordNames.toString());
-
+    super.writeHeader(coords, (StationTimeSeriesCollectionImpl) stations, null);
     int count = 0;
-    stationIndexMap = new HashMap<>(2 * stns.size());
+    stationIndexMap = new HashMap<>(2 * stnList.size());
     for (StationFeature stn : stnList) {
```

Comment:
an same question here

Reply:
I'm not 100% sure, I just changes which collection was getting looped over so I changed the initialCapacity too. If I had to guess, the load factor is .75 so setting the initialCapacity to (2 * stnList.size()) guarantees that the hashmap won't get resized.